### DISCUSSION TITLE
Webdriverio 7 WebdriverIO.Config is not recognised

### DISCUSSION THREAD

**Discussion by bobige** [webdriverio/webdriverio#6636] [Q&A] [answered]
Hi guys,

I have a similar problem to that in this discussion https://github.com/webdriverio/webdriverio/discussions/6502.

I updated from v6 to v7 and the import for Config cannot be found:

const baseConfig: WebdriverIO.Config = { ... }

"Namespace 'WebdriverIO' has no exported member 'Config'.ts(2694)"

Any help would be appreciated. Thanks in advance.

**Comment by christian-bromann** [comment, marked as answer]
Please make sure you [setup TypeScript](https://webdriver.io/docs/typescript) properly, especially the part about adding webdriverio to the `types` property in your `tsconfig.json`.

**Reply by bobige** [reply]
Hello, I think that everything should be fine there. This is my `tsconfig.json`:
```
{
  "compilerOptions": {
      "target": "ES2018",
      "baseUrl": ".",
      "paths": {
          "*": [ "./*" ],
      },
      "types": [
          "node",
          "webdriverio/sync",
          "@wdio/mocha-framework", 
          "@wdio/devtools-service", 
          "./types/wdio",
          "expect-webdriverio",
        ]
  },    
  "include": [
      "./.eslintrc.js",
      "./src/**/*.ts",
      "./config/**/*.ts",
      "./test/**/*.ts"
  ],
  "exclude": [
    "./node_modules"
  ]
}
```

**Reply by christian-bromann** [reply]
Please have a look in on of our boilerplates where this seems to work fine: https://github.com/webdriverio/jasmine-boilerplate/blob/main/wdio.sauce.conf.ts

**Reply by FrederickEngelhardt** [reply]
Regarding the usage of `types` if we go this direction, we now have identify each library that is using a types declaration and include them. For large projects that might not be a realistic option.

Is there a better way to do this? Attempted to get this lib specifically *webdriverio** working with `typeRoots` but no luck. Example being 

```
"typeRoots": [
      "./node_modules/@types", // works
      "./node_modules/webdriverio", // does not work
      "./node_modules/webdriverio/sync", // does not work
       "node_modules/webdriverio/sync.d.ts", // does not import the sync.d.ts file
      "./node_modules/@wdio/appium-service",
      "./node_modules/@wdio/appium-service/build",
      "./node_modules/@wdio/appium-service/build/index.d.ts",
    ]
```

The above works for the @wdio library but fails to work for `driver` type definitions found in `node_modules/webdriverio/sync.d.ts`

**Reply by christian-bromann** [reply]
> Regarding the usage of types if we go this direction, we now have identify each library that is using a types declaration and include them. For large projects that might not be a realistic option.

Can you elaborate on this? Why should this not work for large projects?
